edx_sga/sga.py

```diff
@@ -0,0 +1,530 @@
+                    'annotated': state.get("annotated_filename"),
+                    'comment': state.get("comment", ''),
+                }
+
+        return {
+            'assignments': list(get_student_data()),
+            'max_score': self.max_score(),
+        }
+
+    def studio_view(self, context=None):
+        try:
+            cls = type(self)
+
+            def none_to_empty(x):
+                return x if x is not None else ''
+            edit_fields = (
+                (field, none_to_empty(getattr(self, field.name)), validator)
+                for field, validator in (
+                    (cls.display_name, 'string'),
+                    (cls.points, 'number'),
+                    (cls.weight, 'number'))
+            )
+
+            context = {
+                'fields': edit_fields
+            }
+            fragment = Fragment()
+            fragment.add_content(
+                render_template(
+                    'templates/staff_graded_assignment/edit.html',
+                    context
+                )
+            )
+            fragment.add_javascript(_resource("static/js/src/studio.js"))
+            fragment.initialize_js('StaffGradedAssignmentXBlock')
+            return fragment
+        except:  # pragma: NO COVER
+            log.error("Don't swallow my exceptions", exc_info=True)
+            raise
+
+    @XBlock.json_handler
+    def save_sga(self, data, suffix=''):
+        self.display_name = data.get('display_name', self.display_name)
+        self.weight = data.get('weight', self.weight)
+
+        # Validate points before saving
+        points = data.get('points', self.points)
+        # Check that we are an int
+        try:
+            points = int(points)
+        except ValueError:
+            raise JsonHandlerError(400, 'Points must be an integer')
+        # Check that we are positive
+        if points < 0:
+            raise JsonHandlerError(400, 'Points must be a positive integer')
+        self.points = points
+
+    @XBlock.handler
+    def upload_assignment(self, request, suffix=''):
+        require(self.upload_allowed())
+        upload = request.params['assignment']
+        sha1 = _get_sha1(upload.file)
+        answer = {
+            "sha1": sha1,
+            "filename": upload.file.name,
+            "mimetype": mimetypes.guess_type(upload.file.name)[0],
+        }
+        student_id = self.student_submission_id()
+        submissions_api.create_submission(student_id, answer)
+        path = self._file_storage_path(sha1, upload.file.name)
+        if not default_storage.exists(path):
+            default_storage.save(path, File(upload.file))
+        return Response(json_body=self.student_state())
+
+    @XBlock.handler
+    def staff_upload_annotated(self, request, suffix=''):
+        require(self.is_course_staff())
+        upload = request.params['annotated']
+        module = StudentModule.objects.get(pk=request.params['module_id'])
```

Comment:
This would represent another valid workaround for the shortcomings of the xblock system, but it is still a workaround.  The problem is that there is no API available in XBlock for modifying or reading student state if you aren't the student.  This workaround addresses the problem one way, creating a table would address it another way.